### PR TITLE
Feat/use api gateway

### PR DESCRIPTION
**Pull request by danielstorey**

### CODE REVIEW

File: .github/workflows/release-please.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: google-github-actions/release-please-action@v3
+        id: release
+        with:
+          release-type: node
+          default-branch: main
+          package-name: leviate
+      # The logic below handles the npm publication:
+      - uses: actions/checkout@v2
```
Comment (reeves999999) [inline .github/workflows/release-please.yml:18]: v3 is recommended here as node 12 deprecation warning shows.

File: __tests__/core/useApiGateway.spec.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+    it('should post to the endpoint with a trailing slash', () => {
+      const api = useApiGateway('test-api/');
+
+      api.post(postData);
+
+      expect(instance.post).toHaveBeenCalledWith('test-api', postData, undefined);
+    });
+
+    it('should post to the endpoint with a LEADING AND trailing slash', () => {
```
Comment (reeves999999) [inline __tests__/core/useApiGateway.spec.js:34]: Needs leading slash to be sent.

File: __tests__/core/useApiGateway.spec.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+    it('should join subsequent subpaths with an encoded slash', () => {
+      const api = useApiGateway('test-api');
+
+      api.post('/a/b/c/', postData);
+
+      expect(instance.post).toHaveBeenCalledWith('test-api/a%2Fb%2Fc', postData, undefined);
+    });
+
+    it('should strip a slash if the base ends and the path begins with a slash', () => {
```
Comment (reeves999999) [inline __tests__/core/useApiGateway.spec.js:95]: missing

File: __tests__/core/hostPlugin.spec.js [hunk 4]
```diff
@@ -61,4 +67,19 @@ describe('Host plugin', () => {
...
   })
-})
+
+  describe('localize', () => {
+    it('should translate a string exactly as defined in the locales file', () => {
+      const translated = useLocalize().$l('test_phrase');
+
+      expect(translated).toBe('test phrase');
+    });
+
```
Comment (reeves999999) [inline __tests__/core/hostPlugin.spec.js:77]: extra line

File: __tests__/core/rootStore.spec.js [hunk 5]
```diff
@@ -120,32 +124,30 @@ describe('Root store', () => {
...
+          rootStore.modules.foo().setFoo(4);
 
-          setTimeout(() => {
-            reject();
-          }, 1000);
+          setTimeout(reject, 1000);
         });
       };
 
       // await rootStore.transact(callbackWithError);
```
Comment (reeves999999) [inline __tests__/core/rootStore.spec.js:147]: delete?

**Review by leearaneta — APPROVED** [review]

File: __tests__/core/rootStore.spec.js (hunk 5 above)
Comment (danielstorey) [reply __tests__/core/rootStore.spec.js:147]: This actually needs a bit more work. I'm going to leave it in for now.

**Pull request merged**